### PR TITLE
fix: generate type declarations for build

### LINKED ISSUE #272: No Type Declerations for React build
**Describe the bug**
Cannot find module '@trendyol/baklava/dist/baklava-react' or its corresponding type declarations.

**To Reproduce**
1. Import component from "@trendyol/baklava/dist/baklava-react"
2. See error

**Expected behavior**
Should have declerations and typing support

**Additional context**

### PR DESCRIPTION
**Pull request by olkeoguz**
Includes:
* Generating the types with tsc,
* Fixing the generate-react-exports.js by including all the component type imports and hooking up the component with its types to the "createComponent" function provided by @lit-labs/react
* Updating the docs for the beta version.

Closes #272.

### CODE REVIEW

**Review by muratcorlu — CHANGES_REQUESTED** [review]

File: scripts/generate-react-exports.js [hunk 1]
```diff
@@ -37,11 +40,17 @@ for (const module of customElementsModules) {
       }, {})
     : {};
 
+  const importPath = path.replace(/^src\//, '').replace(/\.ts$/, '');
+  const Component = componentName + 'Component';
+
+  importStatements.push(`import ${Component} from "./${importPath}";`);
```
[the remark this answers is not in the document]
Comment (muratcorlu) [reply scripts/generate-react-exports.js:46]: If you just need types, maybe you can try [import type](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html).

[the remark this answers is not in the document]
Comment (olkeoguz) [reply scripts/generate-react-exports.js:46]: In order to generate the React element, we need to provide our component as the third argument of the createComponent function. 

Currently we are setting this argument as `customElements.get('${fileName}')` and this stays as a string in the generated baklava-react.ts file.

See the description below.

https://github.com/lit/lit/tree/main/packages/labs/react

image

I used npm pack to try our package locally in a CRA and get the error you mentioned.

Maybe we can access our component in different way. Any suggestion?

[the remark this answers is not in the document]
Comment (muratcorlu) [reply scripts/generate-react-exports.js:46]: Maybe we can just set a type for it:

```ts
import type ${Component} from from "./${importPath}";`
...
customElements.get('${fileName}') as ${Component}
```
or
```ts
customElements.get<${Component}>('${fileName}')
```

File: README.md [hunk 2]
```diff
@@ -6,14 +6,18 @@
...
-Baklava is a design system provided by [Trendyol](https://github.com/trendyol) to create a consistent UI/UX for app users. 
+Baklava is a design system provided by [Trendyol](https://github.com/trendyol) to create a consistent UI/UX for app users.
 
 Web implementation of the design system is created as native web components so it can be used within every type of web frameworks including Vue, React or Angular. Our target is providing a UI library that has neatly designed and developed for providing best possible user experience for the users of applications that uses Baklava DS.
 
 ## How to use
 
 Preferred way of using Baklava is using it via CDN. Just import library JS and CSS files to your main document like below:
 
+ <bl-alert variant="warning" icon>
```
[the remark this answers is not in the document]
Comment (olkeoguz) [reply README.md]: Hope to include it soon then 😊 I changed it for now.

**Review by muratcorlu — APPROVED** [review]

File: scripts/generate-react-exports.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (leventozen) [reply scripts/generate-react-exports.js]: We are moving on this by adding type at `createComponent` function like this `createComponent<${Type}>`. After @olkeoguz added this, this conversation will be resolved.

**Review by leventozen — APPROVED** [review]

**Pull request merged**